Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -70,7 +70,14 @@ class Navigator {
      Provides a new or an existing `MapboxCoreNavigation.Navigator` instance. Upon first initialization will trigger creation of `MapboxNavigationNative.Navigator` and `HistoryRecorderHandle` instances,
      satisfying provided configuration (`tilesVersion` and `tilesURL`).
      */
-    static let shared: Navigator = Navigator()
+    static var shared: Navigator {
+        return _navigator
+    }
+    
+    // Used in tests to recreate the navigator
+    static var _navigator: Navigator = .init()
```

Comment:
Used only in tests before the bug in NavNative will be fixed:
https://github.com/mapbox/mapbox-navigation-native/pull/3110